qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/BackupIT.java

```diff
@@ -0,0 +1,115 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer.Service;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+class BackupIT {
+  @Container
+  private static final LocalStackContainer S3 =
+      new LocalStackContainer(DockerImageName.parse("localstack/localstack:0.14.5"))
+          .withServices(Service.S3);
+
+  private GrpcClientRule client;
+
+  @RegisterExtension
+  private final ClusteringRuleExtension clusteringRule =
+      new ClusteringRuleExtension(1, 1, 1, this::configureBackupStore);
+
+  private BackupRequestHandler backupRequestHandler;
+
+  private void configureBackupStore(final BrokerCfg config) {
+    final var backupConfig = config.getData().getBackup();
+    backupConfig.setStore(BackupStoreType.S3);
+
+    final var s3Config = backupConfig.getS3();
+    final String bucketName = RandomStringUtils.randomAlphabetic(10).toLowerCase();
+    s3Config.setBucketName(bucketName);
+    s3Config.setEndpoint(S3.getEndpointOverride(Service.S3).toString());
+    s3Config.setRegion(S3.getRegion());
+    s3Config.setAccessKey(S3.getAccessKey());
+    s3Config.setSecretKey(S3.getSecretKey());
+
+    // Create bucket before for storing backups
+    final var s3ClientConfig =
+        io.camunda.zeebe.backup.s3.S3BackupConfig.from(
+            bucketName,
+            s3Config.getEndpoint(),
+            S3.getRegion(),
+            S3.getAccessKey(),
+            S3.getSecretKey());
+    try (final var s3Client = S3BackupStore.buildClient(s3ClientConfig)) {
+      s3Client.createBucket(builder -> builder.bucket(bucketName).build()).join();
+    }
+  }
+
+  @BeforeEach
+  void setup() {
+    client = new GrpcClientRule(clusteringRule.getClient());
+    backupRequestHandler = new BackupRequestHandler(clusteringRule.getGateway().getBrokerClient());
+  }
+
+  @Test
+  void shouldBackup() {
+    // given
+    client.createSingleJob("test");
+
+    final long backupId = 2;
+
+    // when
+    backup(backupId);
+
+    // then
+    Awaitility.await("Backup must be completed.")
+        .timeout(Duration.ofMinutes(2))
+        .ignoreExceptions()
+        .untilAsserted(
+            () -> {
+              final var status = getBackupStatus(backupId);
+              assertThat(status.status()).isEqualTo(BackupStatusCode.COMPLETED);
+              assertThat(status.backupId()).isEqualTo(backupId);
+              assertThat(status.partitions()).hasSize(clusteringRule.getPartitionCount());
+            });
+  }
+
+  private BackupStatus getBackupStatus(final long backupId)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    // TODO: This should be replaced by the Gateway rest api when it is available
```

Comment:
❓ how do we usually make such such TODOs are not overlooked? is this listed in the corresponding issue for the gateway rest api?

Reply:
There is no specific policy. Adding an integration test will be done as part of the issue, so who ever works on this will see this comment.